src/lib/mina_ledger/sparse_ledger.ml

```diff
@@ -77,3 +78,36 @@ let%test_unit "of_ledger_subset_exn with keys that don't exist works" =
       [%test_eq: Ledger_hash.t]
         (Ledger.merkle_root ledger)
         ((merkle_root sl :> Random_oracle.Digest.t) |> Ledger_hash.of_hash))
+
+module T = Mina_transaction_logic.Make (L)
```

Comment:
I think this transaction logic is needed by snarkyjs for its "Mock blockchain" interface (which is important to maintain as a way for developers to test their zk apps locally). Moving this into mina_ledger makes it inaccessible to snarkyjs.

So, either this has to be moved somewhere it remains accessible for JS compilation, or it's simple enough to just be duplicated for snarkyjs!

Reply:
(Leaving another comment so I can "request changes")

Reply:
> So, either this has to be moved somewhere it remains accessible for JS compilation, or it's simple enough to just be duplicated for snarkyjs!

You should still be able to use it just as before; the only difference is you'll need to add `mina_transaction` in your dune file and `open Mina_transaction` in the `.ml` file

Reply:
Yeah sorry, I was wrong... snarkyjs uses Transaction_logic and not Sparse_ledger_base

Reply:
This realization would have simplified my earlier refactorings to mina_ledger a lot :D